modules/extensions-support/src/main/java/org/mule/module/extension/internal/spi/BaseCapabilityExtractor.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.module.extension.internal.spi;
+
+import org.mule.extension.introspection.declaration.fluent.ConfigurationDescriptor;
+import org.mule.extension.introspection.declaration.fluent.DeclarationDescriptor;
+import org.mule.extension.introspection.declaration.fluent.OperationDescriptor;
+import org.mule.module.extension.spi.CapabilityExtractor;
+
+import java.lang.reflect.Method;
+
+/**
+ * Base implementation of {@link CapabilityExtractor} which provides null implementations for all its
+ * extract methods.
```

Comment:
not really. I want this class to be abstract because it makes no sense to create instances of it. The methods should not be abstract either, because what I want is for extended types to only implement the levels they need. Makes sense??